lib/hunter/commands/send.rb

```diff
@@ -66,20 +66,21 @@ def run
 
           request.body = data.to_json
 
-          loop do
-            response = send_request(http, request)
-            if !retry_interval || response&.code == '200'
-              break
-            end
-            sleep(retry_interval.to_f)
+          if send_request(http, request)&.code != '200' && retry_interval
+            begin
+              sleep(retry_interval.to_f)
+            end while send_request(http, request)&.code != '200'
```

Comment:
The `while` block being stuck on the same line as the `end` here is a bit jarring (see [style guide](https://github.com/rubocop/ruby-style-guide#loop-with-break)). Also, there's no need for a begin...end block here. The following is slightly better styling:
```ruby
request.body = data.to_json

return unless retry_interval

until send_request(http, request)&.code == '200'
  sleep(retry_interval.to_f)
end
```

Reply:
but there should be at least one `send_request` call

Reply:
There is. From the [while loop docs](https://ruby-doc.org/core-2.6/doc/syntax/control_expressions_rdoc.html#label-while+Loop):

> The condition a < 10 is checked before the loop is entered, then the body executes, then the condition is checked again. When the condition results in false the loop is terminated.

Reply:
There is. From the [until loop docs](https://ruby-doc.org/core-2.6/doc/syntax/control_expressions_rdoc.html#label-until+Loop):

> Like a while loop the condition a > 10 is checked when entering the loop and each time the loop body executes. If the condition is false the loop will continue to execute.

Reply:
Yeah but I mean if `unless retry_interval` is called first then if it is not enabled (nil), the method just terminated without an initial send action

Reply:
```
return unless send_request(http, request)&.code != '200' && retry_interval

loop do
  sleep(retry_interval.to_f)
  break unless send_request(http, request)&.code != '200'
end
```
I'm gonna do something like this. Well it's true that it looks not graceful to make the `unless send_request()` statement appear twice

Reply:
That does look much better. Good idea :slightly_smiling_face:

Reply:
That is fixable by doing this:

```
loop do
  break if send_request(http, request)&.code == '200' || !retry_interval
  sleep(retry_interval.to_f)
end
```

This way, the loop is broken if we receive a successful response, or the `retry_interval` option is not being used.

Reply:
Yes from a coding perspective, it looks great. But there is an underlying reason I check the retry_interval condition outside the loop is that I hope it won't be called in every loop, which might speed up the program for about 0.0001 seconds. But I have nothing against your implementation. So what's your opinion?

Reply:
You're using a conditional assignment when setting `@retry_interval`, which means that it isn't being calculated every time. It's set the first time, and then recalled each time after that. I don't think that it's worth making the code less readable for such a small optimisation.